source/models/table.py

```diff
@@ -0,0 +1,46 @@
+from abc import ABC
+from dataclasses import dataclass
+import json, os
+from commands_functions.schema_keys import Keys
+from models.table_metadata import TableMetaData
+from outputs.exceptions import *
+
+
+class Table:
+    def __init__(self, database , table_name = None, table_schema = None):
+        self.database = database
+        if table_schema is not None:
+            self.__init_by_schema(table_schema = table_schema)
+        elif table_name is not None:
+            self.__init_by_name(table_name = table_name)
+        else :
+            raise WrongInput(message = "the table name and table schema data are null")
+        self.table_metadata = TableMetaData(self, table_schema)
```

Comment:
There is a bug here. You pass tabe_schema even if it is null

Reply:
I noticed it while implementing the set function today and changed it in another branch I will copy the change